src/frequenz/sdk/timeseries/ev_charger_pool/_ev_charger_pool.py

```diff
@@ -148,3 +166,106 @@ async def power(self, component_id: int) -> FormulaReceiver:
             EVChargerPowerFormula,
             FormulaGeneratorConfig(component_ids={component_id}),
         )
+
+    async def component_data(self, component_id: int) -> Receiver[EVChargerData]:
+        """Stream 3-phase current values and state of an EV Charger.
+
+        Args:
+            component_id: id of the EV Charger for which data is requested.
+
+        Returns:
+            A receiver that streams objects containing 3-phase current and state of
+                an EV Charger.
+        """
+        if recv := self._status_streams.get(component_id, None):
+            if not recv[0].done():
+                return recv[1].new_receiver()
+            logger.warning("Restarting component_status for id: %s", component_id)
+            output_chan = recv[1]
+        else:
+            output_chan = Broadcast[EVChargerData](
+                f"evpool-component_status-{component_id}"
+            )
+
+        task = asyncio.create_task(
+            self._stream_component_data(component_id, output_chan.new_sender())
+        )
+
+        self._status_streams[component_id] = (task, output_chan)
+
+        return output_chan.new_receiver()
+
+    async def _get_current_streams(
+        self, component_id: int
+    ) -> tuple[Receiver[Sample], Receiver[Sample], Receiver[Sample]]:
+        """Fetch current streams from the resampler for each phase.
+
+        Args:
+            component_id: id of EV Charger for which current streams are being fetched.
+
+        Returns:
+            A tuple of 3 receivers stream resampled current values for the given
+                component id, one for each phase.
+        """
+
+        async def resampler_subscribe(metric_id: ComponentMetricId) -> Receiver[Sample]:
+            request = ComponentMetricRequest(
+                namespace="ev-pool",
+                component_id=component_id,
+                metric_id=metric_id,
+                start_time=None,
+            )
+            await self._resampler_subscription_sender.send(request)
+            return self._channel_registry.new_receiver(request.get_channel_name())
+
+        return (
+            await resampler_subscribe(ComponentMetricId.CURRENT_PHASE_1),
+            await resampler_subscribe(ComponentMetricId.CURRENT_PHASE_2),
+            await resampler_subscribe(ComponentMetricId.CURRENT_PHASE_3),
+        )
+
+    async def _stream_component_data(
+        self,
+        component_id: int,
+        sender: Sender[EVChargerData],
+    ) -> None:
```

Comment:
you might want to add documentation for this function in the commit the function is added

Reply:
done